pkg/controller/volume/protectionutil/utils.go

```diff
@@ -0,0 +1,34 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package protectionutil
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/pkg/util/slice"
+)
+
+// Check if object is candidate to be deleted
+func IsDeletionCandidate(obj metav1.Object, finalizer string) bool {
```

Comment:
instead of passing the finializer, is that possible to use the type of the object to determine what is the finalizer?

Reply:
As is it designed in this PR, caller needs to know the finalizer anyway, because it's the caller who adds it to the object. IMO passing it here as argument is OK.

Unless you want to dig deeper and create single struct / interface that implements generic protection of X when Y exists, instantiated only with different configuration in PV / PVC protection controllers. That would be awesome.